app/models/user.rb

```diff
@@ -1,24 +1,38 @@
 class User < ApplicationRecord
-  has_secure_password
-
+  attr_accessor :remember_token
   before_save :downcase_email
 
   validates :name, presence: true
   VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
   validates :email, presence: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}, uniqueness: true
+  has_secure_password
 
-  private
+  def remember
+    self.remember_token = User.new_token
+    update_attribute :remember_digest, User.digest(remember_token)
+  end
 
+  private
   def downcase_email
     self.email.downcase!
   end
 
-  def User.digest string
-    cost = if ActiveModel::SecurePassword.min_cost
-            BCrypt::Engine::MIN_COST
-           else
-            BCrypt::Engine.cost
-           end
-    BCrypt::Password.create string, cost: cost
+  class << self
+    def new_token
+      SecureRandom.urlsafe_base64
+    end
+
+    def User.digest string
+      cost = if ActiveModel::SecurePassword.min_cost
+              BCrypt::Engine::MIN_COST
+             else
+              BCrypt::Engine.cost
+             end
+              BCrypt::Password.create string, cost: cost
+    end
+  end
+
+  def forget
```

Comment:
Can you log out ???

Reply:
cannot. is it move it out from the class<<self?